src/plugins/dashboard/public/dashboard_container/embeddable/create/create_dashboard.ts

```diff
@@ -504,5 +485,50 @@ export const initializeDashboard = async ({
     );
   });
 
+  // --------------------------------------------------------------------------------------
+  // Set up parentApi.query$
+  // Can not use legacyEmbeddableToApi since query$ setting is delayed
+  // --------------------------------------------------------------------------------------
+  untilDashboardReady().then((dashboardContainer) => {
+    const query$ = new BehaviorSubject<Query | AggregateQuery | undefined>(
+      dashboardContainer.getInput().query
```

Comment:
Are we anticipating `dashboardContainer.getInput().query` to be different from `initialDashboardInput.query`? If not, why do we have this nested in an `untilDashboardReady` call? Is there some race condition here vs `legacyEmbeddableToApi`?

Reply:
no race condition. I was unaware of initialDashboardInput. I will change to just use that.

Reply:
Chatted offline. Need to keep in untilDashboardReady to gain access to dashboardContainer. This allows for setting query$ on dashboardContainer and subscribing to getInput$